session/auth/authprovider/podman.go

```diff
@@ -0,0 +1,62 @@
+package authprovider
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/docker/cli/cli/config/configfile"
+	"github.com/pkg/errors"
+
+	"github.com/moby/buildkit/session"
+)
+
+const (
+	// PodmanConfigFileName is the name of config file
+	PodmanConfigFileName = "auth.json"
+
+	// XDGSubPath is the subpath in the XDG_RUNTIME_DIR directory that contains the podman config file
+	XDGSubPath = "containers"
+)
+
+// NewPodmanAuthProvider provides an attachable that loads the Podman config instead of the hardcoded docker ones. It is
+// More or less a copy of the relevant pieces for loading Docker config files used by the DockerAuthProvider, just using
+// Podmans default config paths & names.
+func NewPodmanAuthProvider(stderr io.Writer) session.Attachable {
+	xdgRuntime := os.Getenv("XDG_RUNTIME_DIR")
```

Comment:
What if this env isn't set?
```suggestion
	xdgRuntime, ok := os.Lookup("XDG_RUNTIME_DIR")
```

Reply:
We would fail to load file at `os.Open` with the File does not exist error; triggering the docker file usage. I can make that more explicit if you like.

Reply:
I think a more-explicit check is good as it'll help users identify the root cause of the error more easily.